File: problems/1572_matrix-diagonal-sum/__main__.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        ret = 0
        for i in range(n):
            ret += mat[i][i] + mat[i][n - 1 - i]
        if n % 2:
            ret -= mat[n // 2][n // 2]
        return ret

File: problems/1572_matrix-diagonal-sum/main.go
package main

func diagonalSum(mat [][]int) int {
    n := len(mat)
    var ret int
    for i := 0; i < n; i++ {
        ret += mat[i][i] + mat[n - 1 - i][i]
    }
    if n % 2 == 1 {
        ret -= mat[n / 2][n / 2]
    }
    return ret
}